lib/modules.nix

```diff
@@ -233,9 +233,7 @@ rec {
       if opt.options ? default && res ? default ||
          opt.options ? example && res ? example ||
          opt.options ? description && res ? description ||
-         opt.options ? apply && res ? apply ||
-         # Accept to merge options which have identical types.
-         opt.options ? type && res ? type && opt.options.type.name != res.type.name
```

Comment:
@nbp It is needed to merge `enums` because two `enum` with different parameters have different names, but I am not very sure of the implications removing this can have.
Is this safe?